code/go/0chain.net/core/util/mpt_node_change_test.go

```diff
@@ -0,0 +1,297 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/0chain/gorocksdb"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChangeCollector_AddChange(t *testing.T) {
+	t.Skip("need protect ChangeCollector against concurrent access")
+	t.Parallel()
+
+	cc := &ChangeCollector{
+		Changes: make(map[string]*NodeChange),
+		Deletes: make(map[string]Node),
+	}
+
+	nn := NewValueNode()
+	cc.Deletes[nn.GetHash()] = nn
+
+	on := NewFullNode(&AState{balance: 2})
+	cc.Changes[on.GetHash()] = &NodeChange{Old: nn}
+
+	type fields struct {
+		Changes map[string]*NodeChange
+		Deletes map[string]Node
+	}
+	type args struct {
+		oldNode Node
+		newNode Node
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "Test_ChangeCollector_AddChange_Nil_Old_Node_OK",
+			fields: fields{
+				Changes: cc.Changes,
+				Deletes: cc.Deletes,
+			},
+			args: args{newNode: nn},
+		},
+		{
+			name: "Test_ChangeCollector_AddChange_OK",
+			fields: fields{
+				Changes: cc.Changes,
+				Deletes: cc.Deletes,
+			},
+			args: args{newNode: nn, oldNode: on},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			cc := &ChangeCollector{
+				Changes: tt.fields.Changes,
+				Deletes: tt.fields.Deletes,
+			}
```

Comment:
cc is defined as a local variable to the t.Run block,  but actually, its `Changes` and `Deletes` fields have TestChangeCollector_AddChange scope. This means that logically it is the same cc in all tests. 

This is confusing to the reader, so either move cc out of the testing loop or make the `Changes` and `Deletes` fields local to the t.Run block. As a further incentive, this won't copy the new mutex field added in PR https://github.com/0chain/0chain/pull/281 leading to confounding race errors.

Reply:
Done.